beta-prompts/beta_prompts/improvement_engine.py

```
@@ -0,0 +1,610 @@
+        else:
+            structure = """<instructions>
+Follow this systematic approach:
+1. **Understanding** - Analyze the request and requirements
+2. **Planning** - Organize the approach and methodology
+3. **Execution** - Implement the solution step by step
+4. **Review** - Verify completeness and quality
+</instructions>"""
+
+        return f"{structure}\n\n{prompt}"
+
+    def _generate_thinking_process(self, prompt: str, task_type: str) -> str:
```

Ruff (0.12.2)

Unused method argument: `prompt`

(ARG002)